### ISSUE TITLE
Issues with mining level tag generation

### ISSUE THREAD

**Issue by MechanosG** [lukebemish/excavated_variants#95]
### What went wrong?

Heya.  Testing the mod for the first time on Quilt.  Dug down a bit in a new world, and immediately found some missing textures on Granite Morkite Ore.  The ore is from Mythic Metals.  Could be more, but decided to make a bug report soon as I saw it.

(btw, on your github report template, you should really specify what the "Version" is talking about.  Minecraft version?  Quilt version?  Excavated Variants version?  Gimma da deets)

In case you need it, MC 1.19.2, Quilt 0.17.5-beta3, Excavated Variants 0.8.1, Steel Reserve 8.1%

### Platform

Quilt

### Version

1.19.2

### Logs and Screenshots

![javaw_lmRFTeBmA7](https://user-images.githubusercontent.com/4360512/193736633-ca10ce17-45f0-4645-a7a1-c94072f5ef23.png)

**Comment by lukebemish** [comment]
Logs please, and I'll take a look. I meant mod version, as I have different mod versions for different MC versions.

**Comment by MechanosG** [comment]
Wow, you're fast.  Alright.  I'm still in-game paused.  Here's my current log.

https://mclo.gs/wiPvwCo

**Comment by lukebemish** [comment]
Alright, I'll check it out. May be a few weeks to fix as I don't have a hard drive right now

**Comment by MechanosG** [comment]
Ouch.  Alright.  Was it a mechanical drive failure?  Just curious - I swear by SSDs now, after having a couple spin up drives fail on me out of the blue.  I'll disable the mod for now, but I'll keep an eye on it.  Thanks for the mod/fast reply, and I hope your your hard drive luck improves.

**Comment by lukebemish** [comment]
Some sort of electrical short, it looks like. You should be able to keep using the mod if you don't mind the missing textures, but otherwise it'll be fixed in a week or so

**Comment by MechanosG** [comment]
Playing around with building a new Quilt modpack (probably my 20th modpack), oddly having a lot of luck having things come together, I really like the idea of having ores blend in with the stone dynamically, but as a consequence of streamlining the modpack building process I've been removing or temp disabling things as I go, while trying to report bugs where I can.  Hopefully that explains why I was overly quick to mention disabling the mod.  I'll definitely be re-enabling it once the textures are working, as I think it's a really good fit.  :)

**Comment by MechanosG** [comment]
Oh, quick mention.  I just noticed..  the mining level of the Granite Morkite ore (with the missing texture) doesn't actually match the mining level of the regular Morkite Ore.  The original being diamond mining level, while the excavated variant is apparently stone level or worse.  I'm able to mine it with a stone pickaxe and the Morkite drops.

**Comment by lukebemish** [comment]
It looks like you have the fabric version of DynAssetGen installed on quilt. That will cause tons of issues. Try with the proper version.

**Comment by lukebemish** [comment]
Going to close this; if the issue still occurs, provide a new log and I'll reopen it

**Comment by MechanosG** [comment]
Oh.  Whoops.  My bad..  that's what I get for relying on PolyMC's mod browser too much.  It seems to struggle with detecting quilt variants of mods.

Switching to the quilt version of DynAssetGen fixed the broken texture (along with a pretty substantial increase in load times - gonna try turning on caching).  However, the mining level is still wrong.  Regular Morkite Ore has a mining level of 2, while the Granite Morkite Ore generated by Excavated Variants appears to have no mining level.  It still requires a pick, but even a wooden pickaxe can mine it.  Is this intended?  Or should Excavated Variants be inheriting mining level as well?

**Comment by lukebemish** [comment]
Can you turn on data caching and share the mining level tags from the generated cache if they exist? That's not intended behavior and I swear I have to fix this every other version

**Comment by MechanosG** [comment]
Sure.  Where would I find the tags at?  Poking around, I found what appears to be a cache at mod_data\dynamic_asset_generator\asset_cache\excavated_variants\blockstates , filled with json for each block.  But nothing inside mentions mining levels, so maybe I'm looking at the wrong thing?  Here's what's inside of granite_morkite_ore.json
```
{
  "variants": {
    "": [
      {
        "model": "excavated_variants:block/granite_morkite_ore0",
        "x": 0,
        "y": 0
      }
    ]
  }
}
```

**Comment by lukebemish** [comment]
You turned on asset caching. Turn on data caching, then delete the mod_data folder, then look and let me know what's in the data cache

**Comment by MechanosG** [comment]
I turned both on at the same time actually.  But I'll try with...  oh wait.

Ok I see.  For some reason, the last time I loaded the game, it didn't create a data_cache folder.  But one exists now.  Maybe some delay in it's creation?  Like, I have to load a world first, or exit the game completely?  Anyway, inside is a "c" folder, "excavated_variants" folder, and "minecraft" folder.  As far as I can tell, I don't see anything related to mining level in the "c" and "excavated_variants" jsons.  I'll post a few examples.

"c\tags\blocks\mozanite_ores.json"
```
{"replace":false,"values":["excavated_variants:red_sandstone_mozanite_ore","excavated_variants:smooth_basalt_mozanite_ore","excavated_variants:granite_mozanite_ore","excavated_variants:dripstone_mozanite_ore","excavated_variants:tuff_mozanite_ore","excavated_variants:calcite_mozanite_ore","excavated_variants:sandstone_mozanite_ore","excavated_variants:diorite_mozanite_ore","excavated_variants:andesite_mozanite_ore"]}
```
"c\tags\blocks\iron_ores.json"
```
{"replace":false,"values":["excavated_variants:red_sandstone_iron_ore","excavated_variants:smooth_basalt_iron_ore","excavated_variants:dripstone_iron_ore","excavated_variants:sandstone_iron_ore"]}
```
"c\tags\blocks\ores\morkite.json"
```
{"replace":false,"values":["excavated_variants:red_sandstone_morkite_ore","excavated_variants:smooth_basalt_morkite_ore","excavated_variants:granite_morkite_ore","excavated_variants:dripstone_morkite_ore","excavated_variants:tuff_morkite_ore","excavated_variants:calcite_morkite_ore","excavated_variants:sandstone_morkite_ore","excavated_variants:diorite_morkite_ore","excavated_variants:andesite_morkite_ore"]}
```

In the folder "minecraft\tags\blocks" I see some files named "needs_stone_tool.json", "needs_iron_tool.json", and "needs_diamond_tool.json".  In the stone tool one, I do see an entry for "excavated_variants:granite_mozanite_ore".  In fact, nearly all ores have been thrown into the stone tool category.  Diamond tool is completely empty, and iron tool only has
```
{"replace":false,"values":["excavated_variants:dripstone_gold_ore","excavated_variants:smooth_basalt_gold_ore","excavated_variants:sandstone_gold_ore","excavated_variants:red_sandstone_gold_ore"]}
```

Is that the info you wanted?  I'll delete my cache after posting this and let it generate again, just in case it does something.

**Comment by lukebemish** [comment]
This looks to be at least in part a mythic metals issue. They use some weirdness to inject their tags into stuff at runtime instead of using datagen; I can try to work on a solution for this, but the best solution is for them to just datagen their tags, which I can't fix on my end

**Comment by lukebemish** [comment]
Yeah, basically nothing I can do on my end as they inject into the tag loader instead of providing a runtime data pack

**Comment by MechanosG** [comment]
Is that the case for other mods as well?  I noticed Modern Industrialization's ores got thrown in the stone tool json as well.  Oh, and...  I have confirmed that I can mine the ores with only a wooden pickaxe.  So they seem to not even be stone tool level.

Random thought, feel free to ignore, but I'm kinda curious now.  Can this mod handle ores above diamond tool level?  I believe Mythic Metals is inspired by the Metallurgy mods, which went beyond diamond tool level.

**Comment by lukebemish** [comment]
Support for ores abovr diamond level is planned, but probably doesn't work now. I'd have to guess that MI is doing something similar to Mythic Metals with their tags, and there's similarly little I can do, unless they gave me an API hook of some sort. If stuff is in the needs_stone_tool tag and still is mineable with a wooden pickaxe, then that sounds like some other mod is messing with stuff. The fact that the needs_iron_tool doesn't have diamond stuff in it makes me feel like something else is messing with stuff - I'll poke at it in a bit; I can think of few things that would cause that, but it may be fixable on my end

**Comment by lukebemish** [comment]
Hmm. Can you turn *off* both asset and data caching, delete mod_data, and restart and then close the game, and then provide me with the new log?

**Comment by lukebemish** [comment]
Just checked and MI seems to be doing normal stuff (though all of their stuff is in needs_stone_tool already), so this looks like something else is going on.

**Comment by MechanosG** [comment]
New cache still generating.  I've walked away from computer several times kek.  Will get you a log soon though.

Edit: Well, technically the cache is turned off, so it's not generating a cache. ...  ...you know what I mean.

**Comment by MechanosG** [comment]
Alright, turned off caches, deleted the entire mod_data folder, started the modpack, loaded a singleplayer test world, confirmed that I could still mine excavated variant ores with a wood pickaxe by mining one, then uploaded the log.

https://mclo.gs/ZnPpiok

**Comment by lukebemish** [comment]
You have an *extremely* outdated version of excavated variants. Please install the latest version, delete your excavated variants config file, delete the globalresources folder, and get me the new log if the issue persists.

**Comment by MechanosG** [comment]
*sigh* Curseforge shows 0.8.1 at the top, as if it's the latest.  PolyMC can't even detect any compatible versions, hence me selecting what Curseforge offered up.  Manually grabbing 1.0.3 now.

**Comment by lukebemish** [comment]
Curseforge shows the newest version *period*, so it shows 1.0.3 for *forge*, while the old 0.8 for quilt is the newest beta version. The quilt one is right there too, if you go to "files". If PolyMC can't see it, you should open a bug report with them as that's a major issue

**Comment by MechanosG** [comment]
![firefox_FIvB5hpJ5z](https://user-images.githubusercontent.com/4360512/193941993-fcb7ab35-2d29-49ed-9708-a725f219954e.png)
The average user..  including me..  isn't going to realize that it shows 1 entry for release, one entry for beta, and one entry for alpha, regardless of mod loader.  I only just now realized.  I simply saw a forge version on top and a quilt version under it, and clicked on quilt.

Also, I was thinking the same thing, about the PolyMC thing.  Already browsing PolyMC's issues.  They seem to be aware of "something" going on with Quilt, but I'm not certain yet if they know that Quilt packs are ignoring quilt mods entirely and instead looking for fabric versions.

**Comment by lukebemish** [comment]
Unfortunately, there's nothing I can do about the first - that's a flaw with curseforge. The official CF launcher would get around it - otherwise, yeah, I'd see what's up with PolyMC that that's not working

**Comment by MechanosG** [comment]
Using version 1.0.3 for Quilt, same exact thing happened.  Turned on the cache, and needs_diamond_tool has no entries, needs_iron_tool still only has some gold things, and the rest are put into needs_stone_tool.  "wthit" shows the created ore as not having a mining level still too, so I assume that means still wooden pickaxe.

**Comment by lukebemish** [comment]
New logs please

**Comment by lukebemish** [comment]
I am having difficulty reproducing this at *all* locally. I have installed BYG, MI, and BetterEnd for a good selection of ore-adding mods and everything seems to be working out, which leads me to believe that another mod on your end is interacting strangely with DynAssetGen. I would try doing a binary search and see if you can find a single other mod for which the issues cease if it is removed; otherwise, I can mess with the mods listed in the logs you've provided a bit, but it is hard to tell. My bets would be on anything that messes with mining levels or resource/data loading.

**Comment by MechanosG** [comment]
Heya.  Sorry, I ended up removing EV and DAG so I could continue with development of the modpack, and haven't been paying attention to Github notifications.  I could re-add them, generate a new world, look for some affected ore, then get you a new log if you still want.  I am still interested.

My modlist is getting rather large though..  trying to narrow down a random mod interaction at this point via binary search, would be theoretically pretty time consuming.  Disable half the mods, hope that I didn't remove a needed library, delete DynAssetGen cache, sit through the long re-caching period on my potato computer (might be 20+/- minutes per attempt, depending on how many ore/stone adding mods are in the enabled half - Athlon II X4 640 bottleneck), check and see if the ores are in correct categories in the new cache, rinse and repeat 'till I've checked all 150+ mods.  I'd like to avoid that route if there's another way lol.

I'm not using any mods that affect mining level that I know of.  I do have LazyDFU, and now DashLoader, which I guess affect data loading.  Although DashLoader wasn't present at the time of making this issue.  I do notice just now, that I was accidentally using another mod aimed at accomplishing the same thing as Excavated Variants, called Universal Ores.  I think I was under the impression that Universal Ores was vanilla only, while Excavated Variants had mod support.  But it seems Universal Ores supports mods as well, as long as they use vanilla style ore generation.  Could that mod somehow have affected Excavated Variants?  I don't see anything else at the moment that seems related.
